channeldb/channel.go

```diff
@@ -2082,9 +2096,17 @@ func SerializeHtlcs(b io.Writer, htlcs ...HTLC) error {
 	}
 
 	for _, htlc := range htlcs {
+		// The onion blob and hltc data are stored as a single var
+		// bytes blob.
+		onionAndExtraData := make(
+			[]byte, lnwire.OnionPacketSize+len(htlc.ExtraData),
+		)
+		copy(onionAndExtraData, htlc.OnionBlob[:])
+		copy(onionAndExtraData[lnwire.OnionPacketSize:], htlc.ExtraData)
+
 		if err := WriteElements(b,
 			htlc.Signature, htlc.RHash, htlc.Amt, htlc.RefundTimeout,
-			htlc.OutputIndex, htlc.Incoming, htlc.OnionBlob[:],
+			htlc.OutputIndex, htlc.Incoming, onionAndExtraData,
```

Comment:
The alternative is to pass `htlc.OnionBlob[:], htlc.ExtraData` here, then `ExtraData` can take the full `66000` space.

Reply:
I don't see how that's backwards compatible? 

The key to doing this _without_ a migration is to read the `OnionBlob` and `ExtraData` as one var bytes blob, so that we can deserialize legacy HTLCs and ones with `ExtraData` in the same way (with no migration).

If we pass two byte slices to `WriteElements`, it'll encode each one's length separately which will break decoding for HTLCs that don't have extra data.

Reply:
oh right, forgot about the migration😂

Reply:
Can tell you haven't have several vivid nightmares about this serialization :')